demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -810,20 +819,31 @@ export class DemoMeetingApp implements AudioVideoObserver, DeviceChangeObserver,
 
   dataMessageHandler(dataMessage: DataMessage): void {
     if (!dataMessage.throttled) {
+      const isSelf = dataMessage.senderAttendeeId === this.meetingSession.configuration.credentials.attendeeId;
+      if (dataMessage.timestampMs <= this.lastReceivedMessageTimestamp) {
+        return;
+      }
+      this.lastReceivedMessageTimestamp = dataMessage.timestampMs;
       const messageDiv = document.getElementById('receive-message') as HTMLDivElement;
-      const messageSpan = document.createElement('span') as HTMLSpanElement;
-      messageSpan.setAttribute('style', 'white-space: pre-wrap; word-break: break-word');
-      const messageNameSpan = document.createElement('span') as HTMLSpanElement;
-      messageNameSpan.setAttribute('style','font-weight: bold');
+      const messageNameSpan = document.createElement('div') as HTMLDivElement;
+      messageNameSpan.classList.add('message-bubble-sender');
       messageNameSpan.innerText = (dataMessage.senderExternalUserId.split('#').slice(-1)[0]);
-      const messageTextSpan = document.createElement('span') as HTMLSpanElement;
-      messageTextSpan.innerText = ':' + dataMessage.text();
-      messageSpan.appendChild(messageNameSpan);
-      messageSpan.appendChild(messageTextSpan);
-      if (messageDiv.innerText) {
-        messageDiv.appendChild(document.createElement('br'));
+      const messageTextSpan = document.createElement('div') as HTMLDivElement;
+      messageTextSpan.classList.add(isSelf ? 'message-bubble-self' : 'message-bubble-other');
+      messageTextSpan.innerHTML = this.markdown.render(dataMessage.text()).replace(/[<]a /g, '<a target="_blank" ');
```

Comment:
So we replace innerHTML with innerText before to avoid HTML injection. Does markdown.render handle it automatically?

Reply:
Yes it is handled by `markdown-it` automatically:

https://github.com/markdown-it/markdown-it/blob/master/docs/security.md